### ISSUE TITLE
[Hunter] Traps arming time.

### ISSUE THREAD

**Issue by Jerrythemagicdog** [ValorenWoW/sunstrider-core#188]
**Description:**
Currently all traps have instant trigger time (unless affected by Scatter Shot for some reason - maybe some code?). According to patch notes the traps should have a 2 second arming time. (Patch 2.0.1 (05-Dec-2006): Traps can be set while in combat, but require a 2 second arming time. **and** Patch 3.0.2 (14-Oct-2008): The arming time has been reduced to 1 second, from 2 seconds for all traps.  - to show that it was 2 sec up until TLK.) 
If the target is not in combat it should not have an arming time. According to the discussion I posted below, what matters is the enemies' combat state.

> The desired situation is if the Hunter is out of combat (after using FD) but his target is in combat the trap should have 2 seconds arming time. Traps should be instant only when both the hunter and his enemy are out of combat.
> 
> Desired:
> 
>     Hunter in combat, victim out of combat - instant arming time.
>     Hunter in combat, victim in combat - 2 seconds arming time.
>     Hunter out of combat, victim out of combat - instant arming.
>     Hunter out of combat, victim in combat - 2 seconds arming time.
> 
> Hunter combat state is irrelevant.

**Current behaviour:**
As described above, currently there is no arming / trigger time on traps at all (outside of weird Scatter Shot interaction which looks to be done on purpose or just walking around the trap. )

**Expected behaviour:**
There should be an arming time for traps on targets who are in combat.

**Steps to reproduce the problem:**

1. Duel someone as a hunter
2. Press Freezing Trap (for example)
3. The trap is triggered instantly (unless under effects of Scatter Shot)

**Sources**:
http://wowwiki.wikia.com/wiki/Trap for patch notes

https://github.com/FeenixServerProject/Archangel_2.4.3_Bugtracker/issues/2081 for a discussion and some hints at how people remember it working

http://wowwiki.wikia.com/wiki/Freezing_Trap (Patch 2.0.1 (05-Dec-2006): Freezing Traps will now cancel combat when cast. Traps can be set while in combat, but require a 2 second arming time.)

https://youtu.be/VhCI3DDvGGU?t=780 - Trap being planted at 13:05 (video time) and the Priest with the Warlock are walking over the trap at 13:06, trap activates at 13:07 by the Felhunter - looks like it goes hand in hand with the patch notes. )

In summary, the consensus seems to be at least in the fact that there should not be instant traps at all. As explained in the discussion above what (probably) should be taken into consideration is the victim's combat state, if not then 2 second arm time no matter what - every hunter skills Scatter Shot in order to secure traps anyway.

**rev. hash/commit:** 
Sunstrider rev. 10283 (2e80439) 2019-01-22 18:43:47 +0100 (master branch) (Unix, DEBUG, Static)
